### PR TITLE
Test entry and incumbent settings object for promise jobs

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/5212 which specifies entry, and throws in incumbent tests for good measure while we're here.

Chrome and Firefox both seem to pass these, which is somewhat surprising.

### CODE REVIEW

File: html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-entry.html [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        });
+      }
+    };
+
+    Promise.resolve(thenable);
+  }, "Thenable resolution");
+
+  done();
+};
+</script>
```
Comment (bzbarsky) [inline html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-entry.html]: Thank you, this is a great start.  Additional things worth testing:

1) When the Promise comes from one global but the function passed to `then` or `catch` comes from a different global.
2) When the Promise comes from one global, the thenable passed to `resolve` from another global (well, maybe it's OK if it's the same one), and the `then` function on the thenable comes from a different global.
3) Whether browsers do the "can run a script" checks based one various entry globals.

Followup is fine for adding those; the third one especially is likely to be a bit of a pain.

File: html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+window.onload = () => {
+  const relevantWindow = frames[0].document.querySelector("#r").contentWindow;
+  const runInResolver = frames[1].runWhatYouGiveMe;
+
+  function setupTest(t) {
+    ++testId;
+    const thisTestId = testId;
+
+    relevantWindow.addEventListener("messagereceived", t.step_func(e => {
+      const [recievedTestId, recievedSourceURL] = e.detail;
```
Comment (bzbarsky) [inline html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html]: "received", both times.

File: html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  }, "Rejection handler on rejected promise");
+
+  async_test(t => {
+    const thisTestId = setupTest(t);
+
+    const p = Promise.reject();
+    frames[0].runWindowPostMessageVeryIndirectlyWithNoUserCode(p, "catch", thisTestId, "*");
+  }, "Rejection handler on rejected promise, using backup incumbent settings object stack");
+
+  // The following test that we derive the incumbent settings object at promise-job time from
```
Comment (bzbarsky) [inline html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html]: s/test/tests/

File: html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      runInResolver(reject);
+    }, 0);
+  }, "Rejection handler on pending-then-rejected promise, using backup incumbent settings object stack");
+
+  async_test(t => {
+    const thisTestId = setupTest(t);
+
+    const thenable = {
+      // No t.step_func because that could change the realms
+      then(f) {
```
Comment (bzbarsky) [inline html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html]: Somewhere around here should we have a test where the `then` function of the thenable is effectively a scripted function from the incumbent global (perhaps `runWindowPostMessageVeryIndirectly` bound to some args)?  Maybe it doesn't matter too much....

**Review by bzbarsky — APPROVED** [review]

File: html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-entry.html (hunk 1 above)
Comment (domenic) [reply html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-entry.html]: Added (1) and (2)

Comment (domenic) [reply html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-entry.html]: Filed https://github.com/web-platform-tests/wpt/issues/21756 for (3); thoughts appreciated.

File: html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html (hunk 4 above)
Comment (domenic) [reply html/webappapis/scripting/processing-model-2/integration-with-the-javascript-job-queue/promise-job-incumbent.html]: I'm not sure I quite understand what this adds over the (2) suggested above (where the `then` function is from a different realm). Is using the incumbent global in particular important?

**Pull request merged**